pkg/controllers/loadbalancer/loadbalancer.go

```diff
@@ -217,22 +218,35 @@ func (l *LoadBalancerController) EnsureLoadBalancerDeleted(ctx context.Context,
 	if err != nil {
 		return err
 	}
+
+	l.ipUpdateMutex.Lock()
+	defer l.ipUpdateMutex.Unlock()
+
 	for _, ip := range ips {
-		newTags, last := l.removeServiceTag(*ip, serviceTag)
-		iu := &metalgo.IPUpdateRequest{
-			IPAddress: *ip.Ipaddress,
-			Tags:      newTags,
-		}
-		newIP, err := l.client.IPUpdate(iu)
+		ip := ip
+		err := retrygo.Do(
+			func() error {
+				newTags, last := l.removeServiceTag(*ip, serviceTag)
+				iu := &metalgo.IPUpdateRequest{
+					IPAddress: *ip.Ipaddress,
+					Tags:      newTags,
+				}
+				newIP, err := l.client.IPUpdate(iu)
+				if err != nil {
+					return fmt.Errorf("could not update ip with new tags: %w", err)
+				}
+				l.logger.Printf("updated ip: %v", newIP)
+				if *ip.Type == metalgo.IPTypeEphemeral && last {
+					err := metal.FreeIP(l.client, *ip.Ipaddress)
```

Comment:
Can we add another log line here?

```suggestion
					l.logger.Printf("freeing unused ephemeral ip: %s, tags: %s", *ip.Ipaddress, newTags)
					err := metal.FreeIP(l.client, *ip.Ipaddress)
```